pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -314,89 +330,63 @@ func getPreFilterState(cycleState *framework.CycleState) (*preFilterState, error
-	node := nodeInfo.Node()
+func satisfiesExistingPodsAntiAffinity(state *preFilterState, nodeInfo *framework.NodeInfo) bool {
 	topologyMap := state.topologyToMatchedExistingAntiAffinityTerms
 
 	// Iterate over topology pairs to get any of the pods being affected by
 	// the scheduled pod anti-affinity terms
-	for topologyKey, topologyValue := range node.Labels {
+	for topologyKey, topologyValue := range nodeInfo.Node().Labels {
 		if topologyMap[topologyPair{key: topologyKey, value: topologyValue}] > 0 {
-			klog.V(10).Infof("Cannot schedule pod %+v onto node %v", pod.Name, node.Name)
-			return false, nil
-		}
-	}
-	return true, nil
-}
-
-//  nodeMatchesAllAffinityTerms checks whether "nodeInfo" matches all affinity terms of the incoming pod.
-func nodeMatchesAllAffinityTerms(nodeInfo *framework.NodeInfo, state *preFilterState) bool {
-	node := nodeInfo.Node()
-	for _, term := range state.podInfo.RequiredAffinityTerms {
-		if topologyValue, ok := node.Labels[term.TopologyKey]; ok {
-			pair := topologyPair{key: term.TopologyKey, value: topologyValue}
-			if state.topologyToMatchedAffinityTerms[pair] <= 0 {
-				return false
-			}
-		} else {
 			return false
 		}
 	}
 	return true
 }
 
-//  nodeMatchesAnyTopologyTerm checks whether "nodeInfo" matches any of the pod's anti affinity terms.
-func nodeMatchesAnyAntiAffinityTerm(nodeInfo *framework.NodeInfo, state *preFilterState) bool {
+//  satisfiesPodsAntiAffinity checks if the node satisifies the incoming pod's anti-affinity rules.
+func satisfiesPodsAntiAffinity(state *preFilterState, nodeInfo *framework.NodeInfo) bool {
 	node := nodeInfo.Node()
 	for _, term := range state.podInfo.RequiredAntiAffinityTerms {
 		if topologyValue, ok := node.Labels[term.TopologyKey]; ok {
 			pair := topologyPair{key: term.TopologyKey, value: topologyValue}
 			if state.topologyToMatchedAntiAffinityTerms[pair] > 0 {
-				return true
+				return false
 			}
 		}
 	}
-	return false
+	return true
 }
 
-// getMatchingAntiAffinityTopologyPairs calculates the following for "existingPod" on given node:
-// (1) Whether it has PodAntiAffinity
-// (2) Whether ANY AffinityTerm matches the incoming pod
-func getMatchingAntiAffinityTopologyPairsOfPod(newPod *v1.Pod, existingPod *framework.PodInfo, node *v1.Node) topologyToMatchedTermCount {
-	topologyMap := make(topologyToMatchedTermCount)
-	for _, term := range existingPod.RequiredAntiAffinityTerms {
-		if schedutil.PodMatchesTermsNamespaceAndSelector(newPod, term.Namespaces, term.Selector) {
-			if topologyValue, ok := node.Labels[term.TopologyKey]; ok {
-				pair := topologyPair{key: term.TopologyKey, value: topologyValue}
-				topologyMap[pair]++
+// satisfiesPodsAffinity checks if the node satisfies the incoming pod's affinity rules.
+func satisfiesPodsAffinity(state *preFilterState, nodeInfo *framework.NodeInfo) bool {
+	toposExist := true
+	podsExist := true
+	for _, term := range state.podInfo.RequiredAffinityTerms {
```

Comment:
I'd prefer to refactor the whole function to return early when there is a simple negative path:

```go
func satisfiesPodsAffinity(state *preFilterState, nodeInfo *framework.NodeInfo) bool {
	podsExist := true
	for _, term := range state.podInfo.RequiredAffinityTerms {
		topologyValue, ok := nodeInfo.Node().Labels[term.TopologyKey]
		if !ok {
			return false
		}
		pair := topologyPair{key: term.TopologyKey, value: topologyValue}
		if state.topologyToMatchedAffinityTerms[pair] <= 0 {
			podsExist = false
			break
		}
	}

	if podsExist {
		return true
	}

	// This pod may be the first pod in a series that have affinity to themselves. In order
	// to not leave such pods in pending state forever, we check that if no other pod
	// in the cluster matches the namespace and selector of this pod, the pod matches
	// its own terms, and the node has all the requested topologies, then we allow the pod
	// to pass the affinity check.
	podInfo := state.podInfo
	if len(state.topologyToMatchedAffinityTerms) == 0 && podMatchesAllAffinityTerms(podInfo.Pod, podInfo.RequiredAffinityTerms) {
		return true
	}
	return false
}
```

Reply:
there should be one or two affinity terms and so exiting early is not really beneficial and makes the code more complicated, I prefer simplicity in this case.

Reply:
Or, we can just eliminate the variable `toposExist`  - directly return false in L367, and the logic afterwards would be only checking `podsExist`.

Reply:
done.